packages/core/src/requestPool/requestPoolManager.ts

```diff
@@ -91,14 +91,17 @@ class RequestPoolManager {
    * of the request types, is created. Maximum number of requests of each type
    * is set to 6.
    */
-  constructor() {
+  constructor(id?: string) {
```

Comment:
Having a default id is pretty dangerous. If one isn't provided we should just autogenerate one?